packages/flutter/lib/src/material/circle_avatar.dart

```diff
@@ -95,13 +103,24 @@ class CircleAvatar extends StatelessWidget {
   /// The background image of the circle. Changing the background
   /// image will cause the avatar to animate to the new image.
   ///
+  /// Typically used as a fallback image for [foregroundImage].
+  ///
   /// If the [CircleAvatar] is to have the user's initials, use [child] instead.
   final ImageProvider? backgroundImage;
 
+  /// The foregroundImage image of the circle.
+  ///
+  /// Typically used as profile image. For fallback use [backgroundImage].
+  final ImageProvider? foregroundImage;
+
   /// An optional error callback for errors emitted when loading
   /// [backgroundImage].
   final ImageErrorListener? onBackgroundImageError;
 
+  /// An optional error callback for errors emitted when loading
+  /// [foregroundColor].
+  final ImageErrorListener? onForegroundImageError;
```

Comment:
At line 121, I think the property name should be `[foregroundImage]` not `[foregroundColor]` unless you mean something else in the `ImageErrorListener` that is called `onForegroundImageError`